lib/dialects/better-sqlite3/index.js

```diff
@@ -31,7 +33,31 @@ class Client_BetterSQLite3 extends Client_SQLite3 {
     const bindings = this._formatBindings(obj.bindings);
 
     if (statement.reader) {
-      const response = await statement.all(bindings);
+      if(this.connectionSettings.supportBigNumbers) {
+        statement.safeIntegers(true)
+      }
+      let response = await statement.all(bindings);
+      if(this.connectionSettings.supportBigNumbers) {
+        const bigNumberStrings = this.connectionSettings.bigNumberStrings
+        const processBigintModeRow = (row) => {
```

Comment:
Can be simplified with:

```js
const processBigintModeRow = (row) => {
  return mapValues(row, v => {
    if (typeof v === 'bigint') {
      const numV = Number(v);
      if (numV === v) {
        return numV;
      } else if (bigNumberStrings) {
        return v.toString()
      }
    }
    return v
  })
}
```

Reply:
You are right, this function can be simplified. However, there may be a problem with 'numV === v' because 1n === 1 is false.